### ISSUE TITLE
netInitialize (from Keil Network middleware) fails with NET_ERROR_CONFIG, same code unaffected in API1

### ISSUE THREAD

**Issue by kenshin1988** [ARM-software/CMSIS_5#134]
With API 1, the same code seems to work, but with usage of RTX5, the netinitialize returns with a NET_ERROR_CONFIG.

**Comment by RobertRostohar** [comment]
This is not really about APIs but rather about RTX configuration required for Network middleware. Most likely not enough memory or stack configured in RTX5. You should enable debug in Network to get more information why Network initialization fails.

**Comment by RobertRostohar** [comment]
Cross post: http://www.keil.com/forum/60800/

**Comment by RobertRostohar** [comment]
You should check the Keil forum (see previous comment) or contact Keil technical support:http://www.keil.com/support

**Comment by furbanc** [comment]
RTOS resource requirements are documented here:
http://www.keil.com/pack/doc/mw/Network/html/nw_resource_requirements.html#nw_rtos_req

**Comment by mrfuchs** [comment]
Maybe related: With my project, netInitialize() crashes because of os_svcTimerNew() being called with attr->cb_mem pointing to invalid memory space. In consequence, when the timer control block is tried to be initialized a bus fault occurs. It works like a charm when using RTOS v1.

**Comment by mrfuchs** [comment]
For the record, it seems to work for me when switching from the "IPv4 Debug STDIO" to the "IPv4 Debug" stack.

**Comment by FerdinandEn** [comment]
I have reported this bug to Keil. It will be solved in version 7.4.0 of Middleware.
The reason is that `osTimerDef()` is part of compiled network library. My workaround is a modification of `cmsis_os_v1.c `
Change
```
osTimerId osTimerCreate (const osTimerDef_t *timer_def, os_timer_type type, void *argument) {

  if (timer_def == NULL) {
    return NULL;
  }
// return osTimerNew((osTimerFunc_t)timer_def->ptimer, type, argument, &timer_def->attr);
return osTimerNew((osTimerFunc_t)timer_def->ptimer, type, argument, NULL);
}
```

**Comment by ReinhardKeil** [comment]
Middleware 7.4.0 is now available
